File: backend/src/main/java/com/birdkingdom/entity/BirdLog.java
package com.birdkingdom.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "bird_logs")
public class BirdLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bird_id", nullable = false)
    private Bird bird;

    @Column(name = "log_date", nullable = false)
    private LocalDate logDate;

    @Column(precision = 5, scale = 2)
    private BigDecimal weight;

    @Column(name = "feed_amount", precision = 5, scale = 2)
    private BigDecimal feedAmount;

    @Column(name = "water_amount", precision = 5, scale = 2)
    private BigDecimal waterAmount;

    @Column(length = 20)
    private String mood;

    @Column(columnDefinition = "TEXT")
    private String behavior;

    @Column(name = "is_molting")
    private Boolean isMolting;

    @Column(name = "is_breeding")
    private Boolean isBreeding;

    @Column(precision = 4, scale = 1)
    private BigDecimal temperature;

    @Column(precision = 4, scale = 1)
    private BigDecimal humidity;

    @Column(name = "is_cleaned")
    private Boolean isCleaned;

    @Column(name = "health_score")
    private Integer healthScore;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public BirdLog() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Bird getBird() { return bird; }
    public void setBird(Bird bird) { this.bird = bird; }
    public LocalDate getLogDate() { return logDate; }
    public void setLogDate(LocalDate logDate) { this.logDate = logDate; }
    public BigDecimal getWeight() { return weight; }
    public void setWeight(BigDecimal weight) { this.weight = weight; }
    public BigDecimal getFeedAmount() { return feedAmount; }
    public void setFeedAmount(BigDecimal feedAmount) { this.feedAmount = feedAmount; }
    public BigDecimal getWaterAmount() { return waterAmount; }
    public void setWaterAmount(BigDecimal waterAmount) { this.waterAmount = waterAmount; }
    public String getMood() { return mood; }
    public void setMood(String mood) { this.mood = mood; }
    public String getBehavior() { return behavior; }
    public void setBehavior(String behavior) { this.behavior = behavior; }
    public Boolean getIsMolting() { return isMolting; }
    public void setIsMolting(Boolean isMolting) { this.isMolting = isMolting; }
    public Boolean getIsBreeding() { return isBreeding; }
    public void setIsBreeding(Boolean isBreeding) { this.isBreeding = isBreeding; }
    public BigDecimal getTemperature() { return temperature; }
    public void setTemperature(BigDecimal temperature) { this.temperature = temperature; }
    public BigDecimal getHumidity() { return humidity; }
    public void setHumidity(BigDecimal humidity) { this.humidity = humidity; }
    public Boolean getIsCleaned() { return isCleaned; }
    public void setIsCleaned(Boolean isCleaned) { this.isCleaned = isCleaned; }
    public Integer getHealthScore() { return healthScore; }
    public void setHealthScore(Integer healthScore) { this.healthScore = healthScore; }
    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend_flutter/lib/models/bird_log.dart
class BirdLog {
  final int id;
  final int birdId;
  final String birdName;
  final DateTime logDate;
  final double? weight;
  final double? feedAmount;
  final double? waterAmount;
  final String? mood;
  final String? behavior;
  final bool? isMolting;
  final bool? isBreeding;
  final double? temperature;
  final double? humidity;
  final bool? isCleaned;
  final int? healthScore;
  final String? notes;
  final DateTime? createdAt;

  BirdLog({
    required this.id,
    required this.birdId,
    required this.birdName,
    required this.logDate,
    this.weight,
    this.feedAmount,
    this.waterAmount,
    this.mood,
    this.behavior,
    this.isMolting,
    this.isBreeding,
    this.temperature,
    this.humidity,
    this.isCleaned,
    this.healthScore,
    this.notes,
    this.createdAt,
  });

  factory BirdLog.fromJson(Map<String, dynamic> json) {
    return BirdLog(
      id: json['id'],
      birdId: json['birdId'],
      birdName: json['birdName'] ?? '',
      logDate: DateTime.parse(json['logDate']),
      weight: json['weight']?.toDouble(),
      feedAmount: json['feedAmount']?.toDouble(),
      waterAmount: json['waterAmount']?.toDouble(),
      mood: json['mood'],
      behavior: json['behavior'],
      isMolting: json['isMolting'],
      isBreeding: json['isBreeding'],
      temperature: json['temperature']?.toDouble(),
      humidity: json['humidity']?.toDouble(),
      isCleaned: json['isCleaned'],
      healthScore: json['healthScore'],
      notes: json['notes'],
      createdAt:
          json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }

  String get moodText {
    switch (mood) {
      case 'HAPPY':
        return '开心';
      case 'NORMAL':
        return '正常';
      case 'QUIET':
        return '安静';
      case 'ANXIOUS':
        return '焦虑';
      default:
        return mood ?? '';
    }
  }

  String get summary {
    final parts = <String>[];
    if (weight != null) parts.add('体重${weight}g');
    if (mood != null) parts.add(moodText);
    if (notes != null && notes!.isNotEmpty) parts.add(notes!);
    return parts.join(' · ');
  }
}
